Packs/ShadowxSOCAI/Integrations/ShadowxSOCAI/ShadowxSOCAI.py

```diff
@@ -0,0 +1,905 @@
+import re
+import json
+import time
+from urllib.parse import urljoin, urlparse
+from typing import Any, Dict, Optional
+
+import requests
+
+import demistomock as demisto
+from CommonServerPython import *  # noqa: F401,F403
+from CommonServerUserPython import *  # noqa: F401,F403
+
+
+# ============================
+# Helpers
+# ============================
+
+def _verify_flag() -> bool:
+    """
+    XSOAR 'insecure' means: do NOT verify TLS.
+    So if insecure=true, verify=False -> return not insecure.
+    """
+    return not bool(demisto.params().get("insecure", False))
+
+
+def _new_session() -> requests.Session:
+    """
+    Create a fresh requests.Session per command. Also inject XSOAR proxy settings.
+    """
+    s = requests.Session()
+    proxies = handle_proxy()
+    if proxies:
+        s.proxies.update(proxies)
+    return s
+
+
+def _require_param(name: str, value: Optional[str]):
+    if not value:
+        raise DemistoException(f"Missing required instance parameter: '{name}'")
+
```

Comment:
no need for this function, if you define the param as required in the yml, the instance will not be saved if it's not defined